src/HostManager.cpp

```diff
@@ -214,8 +214,8 @@ void HostManager::postInterHostEvent( const Host * pHost, const TEvent & event )
     allValidHosts = afterSendingHost;
     allValidHosts.append( beforeSendingHost );
 
-    for( int j = 0, total = allValidHosts.size(); j < total; ++j ) {
-        hostList.at( allValidHosts.at( j ) )->raiseEvent( event );
+    for(int allValidHost : allValidHosts) {
```

Comment:
Yeah, it seems good, the intent is to take the list of hosts, e.g. {A, B, C, D, E, F} and, if say D is the sender of the event, to get allValidHosts to contain {E, F, A, B, C} and to send the event to those hosts in that order.